addnewcitydialog.cpp

```diff
@@ -0,0 +1,82 @@
+#include "addnewcitydialog.h"
+#include "ui_addnewcitydialog.h"
+
+#include <QIcon>
+
+AddNewCityDialog::AddNewCityDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::AddNewCityDialog),
+    m_messageBoxInfo{new QMessageBox(this)}
+{
+    ui->setupUi(this);
+}
+
+AddNewCityDialog::~AddNewCityDialog()
+{
+    delete ui;
+}
+
+void AddNewCityDialog::on_save_cancel_buttonBox_accepted()
+{
+    if (ui->inputContinentComboBox->currentText() == "Select continent"
+        || ui->inputCountryLineEdit->text().isEmpty()
+        || ui->inputCityLineEdit->text().isEmpty()
+        || ui->inputPopulationLineEdit->text().isEmpty()
+        || ui->inputHighTempLineEdit->text().isEmpty()
+        || ui->inputLowTempLineEdit->text().isEmpty())
+    {
+        m_messageBoxInfo->setIcon(QMessageBox::Critical);
```

Comment:
Get rid off m_messageBoxInfo and create one when you need it

Reply:
done